src/main/java/org/opensearch/sdk/sample/helloworld/rest/RestHelloAction.java

```diff
@@ -50,24 +55,49 @@ public class RestHelloAction extends BaseExtensionRestHandler {
     private List<String> worldAdjectives = new ArrayList<>();
     private Random rand = new Random();
 
+    /**
+     * Instantiate this action
+     */
+    public RestHelloAction() {}
+
     @Override
-    public List<RouteHandler> routeHandlers() {
+    public List<NamedRoute> routes() {
         return List.of(
-            new RouteHandler(GET, "/hello", handleGetRequest),
-            new RouteHandler(POST, "/hello", handlePostRequest),
-            new RouteHandler(PUT, "/hello/{name}", handlePutRequest),
-            new RouteHandler(DELETE, "/goodbye", handleDeleteRequest)
+            new NamedRoute.Builder().method(GET)
+                .path("/hello")
+                .handler(handleGetRequest)
+                .uniqueName(routePrefix("greet"))
```

Comment:
One last comment, it looks like an extensions developer has the ability to pass any name as unique name and conventionally they would opt to use this `routePrefix`. I'm wondering if it should be stronger than that and that all names for routes with an extension must be prefixed with the extension's name/abbreviation?

Reply:
I think we should leave this responsibility to extensions developer. We can maybe in a future iteration enforce strict checks to validate the prefixes with extension name.